amd_gpu/gpu.c

```diff
@@ -163,10 +163,12 @@ const char* err_to_str(cl_int ret)
 		return "CL_INVALID_LINKER_OPTIONS";
 	case CL_INVALID_DEVICE_PARTITION_COUNT:
 		return "CL_INVALID_DEVICE_PARTITION_COUNT";
+#ifndef __APPLE__
```

Comment:
We'd need to do both as there is no direct support of AMD for Apple's OS. We need to deal with Apple's SDK which seems to be an entirely different thing.

I'll investigate checking for AMD APPSDK version for you on one of my Ubuntu boxes.